sdk/src/main/java/io/opentelemetry/sdk/trace/TraceProtoUtils.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import com.google.common.collect.EvictingQueue;
+import com.google.protobuf.ByteString;
+import io.opentelemetry.proto.trace.v1.Span.Attributes;
+import io.opentelemetry.proto.trace.v1.Span.Links;
+import io.opentelemetry.proto.trace.v1.Span.SpanKind;
+import io.opentelemetry.proto.trace.v1.Span.TimedEvents;
+import io.opentelemetry.proto.trace.v1.TruncatableString;
+import io.opentelemetry.resource.Resource;
+import io.opentelemetry.sdk.internal.TimestampConverter;
+import io.opentelemetry.sdk.trace.RecordEventsSpanImpl.AttributesWithCapacity;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.Tracestate;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+// Utilities to convert to trace protos.
+final class TraceProtoUtils {
```

Comment:
Please add unit tests for this class.

Reply:
I'll post a separate PR on proto utils and tests.